projects/ad7616-st/sdp-ck1z.ioc

```diff
@@ -0,0 +1,193 @@
+#MicroXplorer Configuration settings - do not modify
+CAD.formats=
+CAD.pinconfig=
+CAD.provider=
+File.Version=6
+GPIO.groupedBy=Group By Peripherals
+KeepUserPlacement=false
+Mcu.CPN=STM32F469NIH6
+Mcu.Family=STM32F4
+Mcu.IP0=NVIC
+Mcu.IP1=RCC
+Mcu.IP2=SPI4
+Mcu.IP3=SYS
+Mcu.IP4=UART5
+Mcu.IPNb=5
+Mcu.Name=STM32F469NIHx
+Mcu.Package=TFBGA216
+Mcu.Pin0=PE4
+Mcu.Pin1=PE2
+Mcu.Pin10=PJ5
+Mcu.Pin11=PJ3
+Mcu.Pin12=PA7
+Mcu.Pin13=PJ0
+Mcu.Pin14=PJ1
+Mcu.Pin15=VP_SYS_VS_Systick
+Mcu.Pin2=PB4
+Mcu.Pin3=PB3
+Mcu.Pin4=PC12
+Mcu.Pin5=PE5
+Mcu.Pin6=PE6
+Mcu.Pin7=PD2
+Mcu.Pin8=PH0/OSC_IN
+Mcu.Pin9=PH1/OSC_OUT
+Mcu.PinsNb=16
+Mcu.ThirdPartyNb=0
+Mcu.UserConstants=
+Mcu.UserName=STM32F469NIHx
+MxCube.Version=6.10.0
+MxDb.Version=DB.6.0.100
```

Comment:
MxCube.Version=6.5.0
MxDb.Version=DB.6.0.50
Please use these values instead, these work on the CI

Reply:
I downgraded the version but now I'm getting errors regarding the size of the generated elf segments. I will have a look, but in case it's a known issue here are the logs:
> /usr/bin/../lib/gcc/arm-none-eabi/12.2.1/../../../../arm-none-eabi/bin/ld: /home/esteban/BayLibre/adi/no-os/ad7616_stm32_project/projects/ad7616-st/build/ad7616-st.elf section `.text' will not fit in region `FLASH'
/usr/bin/../lib/gcc/arm-none-eabi/12.2.1/../../../../arm-none-eabi/bin/ld: /home/esteban/BayLibre/adi/no-os/ad7616_stm32_project/projects/ad7616-st/build/ad7616-st.elf section `.data' will not fit in region `RAM'
/usr/bin/../lib/gcc/arm-none-eabi/12.2.1/../../../../arm-none-eabi/bin/ld: region `RAM' overflowed by 9576 bytes
/usr/bin/../lib/gcc/arm-none-eabi/12.2.1/../../../../arm-none-eabi/bin/ld: region `FLASH' overflowed by 109172 bytes

Reply:
strange... I tried it on my machine and for me the build is successful. you may perform the changes and check whether the CI build is also successful. 
I have only encountered similar errors in bigger projects where indeed the available memory was not enough, but it is not the case here.
I don't know what to suggest, make sure the older version packages are downloaded completely before trying to build. you may also try and see if stm32cubemx is able to successfully generate the code using the .ioc with MxCube.Version=6.5.0
MxDb.Version=DB.6.0.50

Reply:
I was not able to open the modified ioc file and there is no straight forward way to downgrade the stm32cubemx version (editing MxCube.Version and MxDb.Version leads to other problems from what I read on Stm32 forums). So I took the ioc file from ad74416h and modified it for this project and I got the exact same issue. I looked at the generated linker script and it's obvious that it's wrong:

```ld
/* Memories definition */
MEMORY
{
  CCMRAM    (xrw)    : ORIGIN = 0x10000000,   LENGTH = 64
  RAM    (xrw)    : ORIGIN = 0x20000000,   LENGTH = 320
  FLASH    (rx)    : ORIGIN = 0x8000000,   LENGTH = 2K
}
```
320 Bytes of RAM? 2K of flash? The [product description](https://www.st.com/en/microcontrollers-microprocessors/stm32f469ni.html) does not agree

Reply:
The CI will not work with versions later than 6.5.0 simply because the STM32CubeMX will give a pop-up window when the project is built, and the CI cannot handle this pop-up. 
So the log file will not be useful, it will simply be a timeout error. 
You can simply modify the ioc in a text editor with the changes I mentioned and push it here to see what the build logs are saying.